docs: add sharded-nothing EN (#79)
apache/incubator-horaedb-docs: docs/src/en/design/shared_nothing.md

# Shared Nothing Architecture

## Background

> In the [article](./clustering.md) on cluster architecture, CeresDB's cluster solution was introduced, which can be

Reviewer: ```suggestion
In the [cluster architecture](./clustering.md), CeresDB's cluster solution is introduced, which can be
```

summarized as follows:

- Computing and storage are separated;
- Managed by a centralized metadata center that oversees the entire cluster;

However, in the architecture of computing and storage separation, an important issue is how to ensure that data in the
shared storage layer will not be corrupted due to different computing nodes accessing it during the cluster scheduling

> process. For example, if the same data block is updated by multiple computing nodes simultaneously, data corruption may

Reviewer: ```suggestion
. For example, if the same data block is updated by multiple computing nodes simultaneously, data corruption may
```

occur.

CeresDB's solution is to achieve a similar effect to
the [Shared-Nothing architecture](https://en.wikipedia.org/wiki/Shared-nothing_architecture) through specific mechanisms
in the
case of shared storage. This means that the data in the storage layer is partitioned according to certain rules, and
**only one CeresDB instance can update it at any given time**. This feature is defined in this article as the
**correctness of
the cluster topology**. If this correctness is ensured, data will not be damaged due to the flexible scheduling of the
cluster.

This article does not elaborate on the pros and cons of the Shared-Nothing architecture. Instead, it mainly shares how

> CeresDB's cluster solution achieves the Shared-Nothing effect (i.e. how to ensure the **correctness of the cluster

Reviewer: ```suggestion
CeresDB's cluster solution implements it (i.e. how to ensure the **correctness of the cluster
```

topology**) in the case of computing and storage separation.

## Data partitioning

> To achieve the effect of Shared Nothing, it is necessary to logically and physically partition the data on the shared

Reviewer: ```suggestion
To implement the Shared-Nothing architecture, it is necessary to partition the data logically and physically on the shared
```

> storage layer. The basic function of Shard was introduced in the previous cluster

Reviewer: ```suggestion
storage layer. The basic function of `Shard` was introduced in the previous cluster
```
